### PR TITLE
TINY-6518: Fixed a security issue where urls in attributes weren't correctly sanitized

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6518

Description of Changes:
* Fixes a security issue reported for certain URLs in attributes.
* Fixed a setting not being passed through correctly.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/api/html/SaxParser.ts [hunk 1]
```diff
@@ -451,7 +459,10 @@ function SaxParser(settings?: SaxParserSettings, schema = Schema()): SaxParser {
             attrList = [];
             attrList.map = {};
 
-            attribsValue.replace(attrRegExp, parseAttribute);
+            attribsValue.replace(attrRegExp, (match, name, val, val2, val3) => {
+              parseAttribute(value, match, name, val, val2, val3);
+              return '';
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/html/SaxParser.ts:464]: This was needed as `replace` needs a string returned, but `parseAttribute` returns `void` so this wasn't technically ever valid previously. It's only flagged now, as I added what the types were.

Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/html/SaxParser.ts:464]: I should also add we could pull this out, but that's more code and going to be a bit slower (SaxParser we want as fast as possible).

File: modules/tinymce/changelog.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@ Version 5.6.0 (TBD)
     Added new `closest` formatter API to get the closest matching selection format from a set of formats. #TINY-6479
     Added new `name` field to the `style_formats` setting object to enable specifying a name for the format. #TINY-4239
     Changed `readonly` mode to allow hyperlinks to be clickable #TINY-6248
+    Fixed a security issue where urls in attributes weren't correctly sanitized #TINY-6518
```
Comment (JamesToohey) [inline modules/tinymce/changelog.txt:8]:
```suggestion
    Fixed a security issue where URLs in attributes weren't correctly sanitized #TINY-6518
```

File: modules/tinymce/src/core/test/ts/browser/html/SaxParserTest.ts [hunk 3]
```diff
@@ -793,6 +794,68 @@ UnitTest.asynctest('browser.tinymce.core.html.SaxParserTest', function (success,
...
+    parser.parse(
+      '<iframe src="data:image/svg+xml;base64,x"></iframe>' +
+      '<a href="data:image/svg+xml;base64,x">1</a>' +
+      '<object data="data:image/svg+xml;base64,x"></object>' +
+      '<img src="data:image/svg+xml;base64,x">' +
+      '<video poster="data:image/svg+xml;base64,x"></video>'
+    );
+    LegacyUnit.equal(
+      writer.getContent(),
+      '<iframe></iframe><a>1</a><object></object>' +
```
Comment (JamesToohey) [inline modules/tinymce/src/core/test/ts/browser/html/SaxParserTest.ts:811]: Can we put the `a` and `object` on separate lines so when reading the test it's clear that the data has been stripped, but the elements remain?

**Review by JamesToohey — APPROVED** [review]

File: modules/tinymce/changelog.txt (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/changelog.txt]: Oops, thanks for catching this!

File: modules/tinymce/src/core/test/ts/browser/html/SaxParserTest.ts (hunk 3 above)
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/html/SaxParserTest.ts]: I did it this way, as that's how the rest of the file works. That being said, I agree it'd be better, so I've reworked the URL checks to use a new line per element.

**Review by ltrouton — APPROVED** [review]

**Review by metricjs — APPROVED** [review]

**Pull request merged**